File: c++/1836.cpp
#include <iostream>
#include <cmath>
#include <cstdio>

using namespace std;

int main() {

    //freopen("1836.in", "r", stdin);

    int t;
    int bs, iv, ev;
    string nome;
    int level;
    string atr;
    int valor;

    cin >> t;

    for(int i = 1; i <= t; i++) {
        cin >> nome >> level;
        cout << "Caso #" << i << ": " << nome << " nivel " << level << endl;
        for(int j = 1; j <= 4; j++) {
            cin >> bs >> iv >> ev;

            if(j == 1) {
                atr = "HP";
            }
            else if(j == 2) {
                atr = "AT";
            }
            else if(j == 3) {
                atr = "DF";
            }
            else if(j == 4) {
                atr = "SP";
            }

            if(j == 1) {
                valor = (((iv + bs + sqrt(ev) / 8 + 50) * level) / 50) + 10;
            } else {
                valor = (((iv + bs + sqrt(ev) / 8) * level) / 50) + 5;
            }
            cout << atr << ": " << valor << endl;
        }
    }
}

File: python3/ad-hoc/1836.py
import math

teste = int(input())

atr = {1: "HP", 2: "AT", 3: "DF", 4: "SP"}

for index in range(1, teste+1):
    nome, level = input().split(" ")
    level = int(level)

    print("Caso #{}: {} nivel {}".format(index, nome, level))

    for index in range(1, 5):
        bs, iv, ev = input().split(" ")
        bs = int(bs)
        iv = int(iv)
        ev = int(ev)

        if(index == 1):
            valor = (((iv + bs + math.sqrt(ev) / 8 + 50) * level) / 50) + 10
        else:
            valor = (((iv + bs + math.sqrt(ev) / 8) * level) / 50) + 5
        
        print("{}: {}".format(atr[index], math.trunc(valor)))
